File: generated-code/swift/codecraft/FileReadWrite/Sources/TransGenTest/Model/Player.swift
/// Player (strategy, client)
public struct Player {
    /// Player's ID
    let id: Int32

    /// Current score
    let score: Int32

    /// Current amount of resource
    let resource: Int32

    /// Read Player from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Player {
        var id: Int32
        id = stream.readInt32()
        var score: Int32
        score = stream.readInt32()
        var resource: Int32
        resource = stream.readInt32()
        return Player(id: id, score: score, resource: resource)
    }

    /// Write Player to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(id)
        stream.writeInt32(score)
        stream.writeInt32(resource)
    }
}

File: generated-code/pascal/codecraft/TcpReadWrite/Model/Player.pas
unit Player;

interface

uses
    Stream,
    SysUtils;

type
    // Player (strategy, client)
    TPlayer = class
        // Player's ID
        id: Int32;
        // Current score
        score: Int32;
        // Current amount of resource
        resource: Int32;
        constructor Create(id: Int32; score: Int32; resource: Int32);
        // Read Player from input stream
        class function ReadFrom(stream: TStream): TPlayer; static;
        // Write Player to output stream
        procedure WriteTo(stream: TStream);
        function ToString: ansistring; override;
    end;

implementation

constructor TPlayer.Create(id: Int32; score: Int32; resource: Int32);
begin
    self.id := id;
    self.score := score;
    self.resource := resource;
end;

class function TPlayer.ReadFrom(stream: TStream): TPlayer;
var id: Int32;
var resource: Int32;
var score: Int32;
begin
    id := stream.ReadInt32;
    score := stream.ReadInt32;
    resource := stream.ReadInt32;
    result := TPlayer.Create(id, score, resource);
end;

procedure TPlayer.WriteTo(stream: TStream);
begin
    stream.WriteInt32(id);
    stream.WriteInt32(score);
    stream.WriteInt32(resource);
end;

function TPlayer.ToString: ansistring;
begin
    result := 'Player {';
    result += 'id=';
    result += IntToStr(id);
    result += ', ';  
    result += 'score=';
    result += IntToStr(score);
    result += ', ';  
    result += 'resource=';
    result += IntToStr(resource);
    result += '}';
end;

end.